pkgs/development/compilers/llvm/common/default.nix

```diff
@@ -80,232 +81,7 @@ let
-                {
-                  after = "19";
-                  path = ../19;
-                }
-                {
-                  after = "15";
-                  before = "19";
-                  path = ../15;
-                }
-                {
-                  before = "15";
-                  path = ../14;
-                }
-              ];
-              "compiler-rt/armv7l.patch" = [
-                {
-                  before = "15";
-                  after = "13";
-                  path = ../13;
-                }
-              ];
-              "compiler-rt/gnu-install-dirs.patch" = [
-                {
-                  before = "14";
-                  path = ../12;
-                }
-                {
-                  after = "13";
-                  before = "15";
-                  path = ../14;
-                }
-                {
-                  after = "15";
-                  before = "17";
-                  path = ../15;
-                }
-                {
-                  after = "16";
-                  path = ../17;
-                }
-              ];
-              "compiler-rt/darwin-targetconditionals.patch" = [
-                {
-                  after = "13";
-                  path = ../13;
-                }
-              ];
-              "compiler-rt/codesign.patch" = [
-                {
-                  after = "13";
-                  path = ../13;
-                }
-              ];
-              "compiler-rt/normalize-var.patch" = [
-                {
-                  after = "16";
-                  path = ../16;
-                }
-                {
-                  before = "16";
-                  path = ../12;
-                }
-              ];
-              "lldb/procfs.patch" = [
-                {
-                  after = "15";
-                  path = ../15;
-                }
-                {
-                  before = "15";
-                  path = ../12;
-                }
-              ];
-              "lldb/resource-dir.patch" = [
-                {
-                  before = "16";
-                  path = ../12;
-                }
-              ];
-              "openmp/fix-find-tool.patch" = [
-                {
-                  after = "17";
-                  before = "19";
-                  path = ../17;
-                }
-              ];
-              "openmp/run-lit-directly.patch" = [
-                {
-                  after = "16";
-                  path = ../16;
-                }
-                {
-                  after = "14";
-                  before = "16";
-                  path = ../14;
-                }
-              ];
-              "libclc/use-default-paths.patch" = [
-                {
-                  after = "19";
-                  path = ../19;
-                }
-              ];
-            };
+            patches = args.patches or import ./patches.nix;
```

Comment:
Thinking aloud: what if this was a function which took (import ./patches.nix) as a parameter? It seems that would allow more easy flexibility for overriding the behaviours.

Are all the patches specified here? My comment is written quick-fire assuming so; it might need more holistic consideration if not.

Reply:
I'm not sure I understand what you're meaning. Mind prototyping the Nix expression?

> Are all the patches specified here?

Yes, we load all patches from the `patches.nix`.

> My comment is written quick-fire assuming so; it might need more holistic consideration if not.

Might be good to have more feedback on this.

Reply:
Another quick fire message.

I'm imagining `patchesFn => patches => patches`:

```nix
{ lib, patchesFn ? lib.id }: { patches = patchesFn (import ./patches); }
```

So that one could pass in as an example, to drop a patch from the package set:
```nix
llvmPackages.override {
  patchesFn = lib.flip builtins.removeAttrs ["clang/gnu-install-dirs.patch"];
}
```

If you wanted to supply an entirely new set of patches:

```nix
llvmPackages.override {
  patchesFn = _: {
    # ... new patches go here ...
  };
}
```

Reply:
Done, thanks for the suggestion. I made it even apply at the toplevel of LLVM so someone could theoretically import & override everything.